Custom options and url params

ryanc16 opened the pull request:
Style options should update in realtime, and url search params should take effect and apply those styles as well.
- [x] need to add tests

cjh15hub committed:
Add Options through Url
- Add AppOptions model
- Add Service to parse params from url
- map shorthand params to full values
- use option values in css
- increase size of ui
- fix horizontal gauges

cjh15hub committed:
Add beginning of Options Menu

cjh15hub committed:
Fix settings display

ryanc16 committed:
Added options menu query param builder

Commit:
Change old dart-sass dependency to newer sass package

Commit:
Change over all css files to scss.
Update options menu styles, data loading, and make it possible to see the changed options in real time.

cjh15hub commented on src/components/options-menu/options-menu.tsx:
```diff
@@ -0,0 +1,208 @@
+      value: props.appOptions.wheelImgUrl
+    },
+    wheelColor: {
+      chooser: (Object.values(ColorOption) as string[]).includes(props.appOptions.wheelColor) ? "predefined" : 'custom',
```
could you use constants or enums instead of repeating those strings? "predefined" and "custom"

ryanc16 replied:
Done: 92b39cdb050

cjh15hub commented on src/components/options-menu/options-menu.tsx:
```diff
@@ -0,0 +1,208 @@
+      [UrlOptions.analogStickFillOpacityKey]: formOptions.analogStickFillOpacity.value,
+      [UrlOptions.analogStickBorderColorKey]: formOptions.analogStickBorderColor.value
+    };
+    return "?" + Object.entries(paramMap).filter(([key, value]) => value != null && value !== '').map(([key, value]) => key + "=" + value).join("&");
```
use !==

cjh15hub replied:
Also is there a utility that basically does this that we can import?

ryanc16 replied:
> use !==

Using `!= null ` will account for null or undefined, and `!== ''` will filter out any empty string values. the only values that can exist here are `undefined` or a string, but I am just trying to avoid any strange behavior with '0' or basically any other falsey values.

> Also is there a utility that basically does this that we can import?

There probably is, but not sure off the top of my head. When I tried looking it up, most things suggest doing it just like we have it, so I'm guessing that's where we got it. One thing I can probably do is use `encodeURIComponent` on the key and value though, which should make it more url safe. 765e66a4ece

cjh15hub replied:
okay, makes sense.

cjh15hub reviewed (commented):
Looks good, just a couple suggestions.

Also:
- [x] Add updated wheel assets. Done: 1471ba4

Commit:
Merge branch 'develop' into feature/custom-options-and-url-params

Commit:
remove backward compatibility nodejs ssl trick

Commit:
Change options menu component to use an enum for predefined or custom color selection method options

Commit:
Adding tests for updated components

Commit:
Use encodeURIComponent on generated custom style url.
Move ColorMethodType enum to top of file so it isn't used before its declaration.

Commit:
Update steering image assets.
Update steering component use the new purple/blue wheel.

Commit:
Combine wheel url and color into a single card, since a custom wheel image needs to be used in place of a predefined colored wheel.
Update wheel color selection down through into the steering component.
